test/lammp/src/test_isprime_ulong.cpp

```
@@ -47,216 +47,477 @@ static const ulong test[140] = {
+                                               288230497383834409,
+                                               288230497383834413,
+                                               288230497383834427,
+                                               288230497383834451,
+                                               288230497383834541};
+
+void test_isprime_uint() {
+    int mix = rand() * rand();
+    bool pass1 = true;
+    auto start = std::chrono::high_resolution_clock::now();
+    for (int i = 0; i < 10000; i++) {
+        uint p = mix + i * 1000000007;
```

Cppcheck (2.20.0)

[error] Signed integer overflow for expression 'i*1000000007'.

(integerOverflow)